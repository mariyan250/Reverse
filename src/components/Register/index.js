// React and Style
import React, { useState } from 'react';
import './index.scss';

// Router
import { Link } from 'react-router-dom';

const Register = () => {
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [rePassword, setRePassword] = useState('');

  const handleRegister = (e) => {
    e.preventDefault();
    alert('Register!');
  }

  return (
    <main className="register">
      <form onSubmit={handleRegister}>
        <h2>Sign Up</h2>

        <div className="row">
          <label htmlFor="name">Full Name</label>
          <input type="text" name="name" value={name}
            onChange={(e) => setName(e.target.value)} required />
        </div>

        <div className="row">
          <label htmlFor="email">Email</label>
          <input type="email" name="email" value={email}
            onChange={(e) => setEmail(e.target.value)} required />
        </div>

        <div className="row">
          <label htmlFor="password">Password</label>
          <input type="password" name="password" value={password}
            onChange={(e) => setPassword(e.target.value)} required />
        </div>

        <div className="row">
          <label htmlFor="rePassword">Repeat Password</label>
          <input type="password" name="rePassword" value={rePassword}
            onChange={(e) => setRePassword(e.target.value)} required />
        </div>

        <button className="auth-button">Submit</button>

        <Link to="/login" className="login-btn">Already have an account?</Link>
      </form>
    </main>
  );
};

export default Register;