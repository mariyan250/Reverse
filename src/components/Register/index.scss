@import '../../index.scss';

.register {
  display: flex;
  flex-direction: column;
  height: 100vh;
  padding: $padding-xxl $padding-l;

  form {
    display: flex;
    flex-direction: column;
    justify-content: center;

    h2 {
      text-align: center;
      font-size: 42px;
      margin: $margin-xl 0 $margin-xxl;
    }

    .auth-button {
      @include authButton;
    }

    .login-btn {
      margin: $margin-l 0 $margin-s;
      align-self: flex-end;
      font-size: 16px;
      text-decoration: none;
      color: $text-black;
    }
  }
}

@media screen and (min-width: 500px) {
  .register {
    max-width: 30rem;
    margin: 0 auto;

    form {
      h2 {
        margin-top: $margin-xs;
      }
    }
  }
}