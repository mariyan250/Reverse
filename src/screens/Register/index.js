// React and style
import React, { useState } from 'react';
import './index.scss';

// Router
import { Link } from 'react-router-dom';

// Components
import Input from 'shared/components/auth/Input';
import Button from 'shared/components/auth/Button';

const Register = () => {
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [rePassword, setRePassword] = useState('');

  const handleRegister = (e) => {
    e.preventDefault();
    alert('Register!');
  };

  return (
    <main className='register vh-100 d-flex flex-column justify-content-center py-5 px-4 my-0 mx-auto'>
      <form
        className='d-flex flex-column justify-content-center'
        onSubmit={handleRegister}>
        <h2 className='text-center mt-4 mb-5 mt-md-2'>Sign Up</h2>

        <section className='mb-4 mt-2'>
          <Input
            name='username'
            label='Full Name'
            type='text'
            onChange={(e) => setName(e.target.value)}
            value={name}
            required
          />
        </section>

        <section className='mb-4'>
          <Input
            name='email'
            label='Email'
            type='email'
            onChange={(e) => setEmail(e.target.value)}
            value={email}
            required
          />
        </section>

        <section className='mb-4'>
          <Input
            name='password'
            label='Password'
            type='password'
            onChange={(e) => setPassword(e.target.value)}
            value={password}
            required
          />
        </section>

        <section className='mb-4'>
          <Input
            name='rePassword'
            label='Repeat Password'
            type='password'
            onChange={(e) => setRePassword(e.target.value)}
            value={rePassword}
            required
          />
        </section>

        <Button text='Submit' />

        <Link
          to='/login'
          className='login-btn mt-4 mb-3 align-self-end text-decoration-none'>
          Already have an account?
        </Link>
      </form>
    </main>
  );
};

export default Register;
