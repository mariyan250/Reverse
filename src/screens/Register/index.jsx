// React and style
import React, { useState, useEffect } from 'react';
import styles from './index.module.scss';

// Redux
import { connect } from 'react-redux';
import { closeHamburger } from 'store/actions/hamburger';

// Router and Routes
import { Link } from 'react-router-dom';
import { routes } from 'constants/routes';

// Translation
import { useTranslation } from 'react-i18next';

// Components
import Input from 'components/auth/Input';
import Button from 'components/auth/Button';

function Register({ closeHamburger }) {
  const [username, setUsername] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [rePassword, setRePassword] = useState('');

  const { t } = useTranslation();

  useEffect(() => {
    closeHamburger();
  }, [closeHamburger]);

  const inputs = [
    {
      name: 'username',
      label: t('register.inputs.labels.username'),
      type: 'text',
      value: username,
      required: true,
      onChange: (e) => setUsername(e.target.value),
      testId: 'usernameInput',
    },
    {
      name: 'email',
      label: t('register.inputs.labels.email'),
      type: 'email',
      value: email,
      required: true,
      onChange: (e) => setEmail(e.target.value),
      testId: 'emailInput',
    },
    {
      name: 'password',
      label: t('register.inputs.labels.password'),
      type: 'password',
      value: password,
      required: true,
      onChange: (e) => setPassword(e.target.value),
      testId: 'passwordInput',
    },
    {
      name: 'rePassword',
      label: t('register.inputs.labels.re_password'),
      type: 'password',
      value: rePassword,
      required: true,
      onChange: (e) => setRePassword(e.target.value),
      testId: 'rePasswordInput',
    },
  ];

  const handleRegister = (e) => {
    e.preventDefault();
  };

  return (
    <main className="px-4">
      <form onSubmit={handleRegister} className={`${styles.form} mx-auto`} data-testid="form">
        <h1 className="text-center mb-5">{t('register.heading')}</h1>

        {inputs.map((input, i) => (
          <div className="mb-4" key={i}>
            <Input {...input} />
          </div>
        ))}

        <div className="mt-5">
          <Button text={t('register.button')} />
        </div>

        <Link to={routes.login} className="app-text-secondary mt-4 d-inline-block">
          {t('register.option_button')}
        </Link>
      </form>
    </main>
  );
}

export default connect(null, { closeHamburger })(Register);
