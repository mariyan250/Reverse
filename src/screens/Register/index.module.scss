@import '../../index';

.form {
  width: to-rem(350px);

  a {
    font-size: to-rem(16px);
  }
}
