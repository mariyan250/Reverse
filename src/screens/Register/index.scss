@import '../../index';

.register {
  form {
    .register-btn {
      font-size: to-rem(16px);
    }
  }
}

@include media-breakpoint-md-up {
  .register {
    max-width: to-rem(400px);
  }
}
