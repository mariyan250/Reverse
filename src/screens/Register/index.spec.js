import React from 'react';
import { renderWithRouter } from 'tests/test-utils';
import Register from './index';

describe('Register', () => {
  let initialState;

  beforeAll(() => {
    initialState = {};
  });

  describe('Rendering', () => {
    it('should render the component with initial state', () => {
      const { asFragment } = renderWithRouter(<Register />, initialState);

      expect(asFragment(<Register />)).toMatchSnapshot();
    });
  });
});
