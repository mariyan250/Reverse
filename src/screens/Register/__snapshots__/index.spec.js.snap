// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Rendering should render the component with initial state 1`] = `
<DocumentFragment>
  <div
    class="_loading_overlay_wrapper _loading_overlay_wrapper--active css-m8dnq0"
    data-testid="wrapper"
  >
    <div
      class="_loading_overlay_overlay css-df17o1"
      data-testid="overlay"
    >
      <div
        class="_loading_overlay_content css-42igfv"
      >
        <div
          class="_loading_overlay_spinner css-50etie"
        >
          <svg
            viewBox="25 25 50 50"
          >
            <circle
              cx="50"
              cy="50"
              fill="none"
              r="20"
              stroke-miterlimit="10"
              stroke-width="2"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
